### ISSUE TITLE
Travis build fails randomly due to Maven dependencies download

### ISSUE THREAD

**Issue by d-gs** [aposin/LicenseScout#20]
## Bug report

For some reason, the Travis builds are starting to fail randomly with the following error after running maven (the dependency might vary):

```
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building LicenseScout Maven Plugin 1.3.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugin-plugin/3.6.0/maven-plugin-plugin-3.6.0.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.052 s
[INFO] Finished at: 2019-06-27T11:30:29Z
[INFO] Final Memory: 9M/34M
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin org.apache.maven.plugins:maven-plugin-plugin:3.6.0 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.apache.maven.plugins:maven-plugin-plugin:jar:3.6.0: Could not transfer artifact org.apache.maven.plugins:maven-plugin-plugin:pom:3.6.0 from/to central (https://repo.maven.apache.org/maven2): Access denied to: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugin-plugin/3.6.0/maven-plugin-plugin-3.6.0.pom , ReasonPhrase:Forbidden. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
The command "mvn -DskipTests=true -Dmaven.javadoc.skip=true install" failed and exited with 1 during .
Your build has been stopped.
```

## Expected Behavior - What where you expecting to happen?

Builds should be able to download from Maven central in Travis.

## Current Behavior - What happens?

Failing builds due to random errors (most likely due to some Travis issue).

## Possible Solution

A possible solution to avoid this in the future is to setup cache for the maven folder in the travis configuration; this will avoid to download directly from Maven central and minimize problems downloading the dependencies.

The drawback is that in some cases this cache might download a corrupted binary in travis and the cache must be manually clean manually by a maintainer. Also, it does not solve 100% of the issue, as it looks like a Travis/Maven problem.

## What are the steps to reproduce this issue? (for bugs)

Non reproducible (just some builds fails in Travis).

## Context

See the main description.

## Any logs, error output, etc?

See the main description.

## Your Environment

* Version used: any
* Environment name and version (e.g. Chrome 39, node.js 5.4): Travis
* Operating System and version (desktop or mobile): linux container

**Comment by d-gs** [comment]
More information found here:
* https://travis-ci.community/t/continuous-maven-repo-403/3908/2
* https://github.com/travis-ci/travis-ci/issues/6593

Looks like a generalized problem. By now, we will do the following actions:
* Disable the build from the "Required" checks (and probably  we will also need to disable Codecov as it runs after successful runs)
* @aposin/maintainers will manually check PRs with code changes (build and code coverage) - it might also be possible to do so by re-run the Travis build until it pass
* Evaluate if the cache functionality could be use in our case (for that, we have to change the `mvn install` to other goal, if not required to check the whole installation.
* Follow up the Travis issue to recover the "Requirements" for PR inclusion into `master`

**Comment by holzerma** [comment]
@d-gs @pfistererm maybe we should switch to Jenkins ?

**Comment by d-gs** [comment]
Looks like the problem was solved at the Travis side. Priority changed to "Medium" to implement the cache in any case (see #21).

**Comment by d-gs** [comment]
Closing: cache in Travis merged (#21).
